source/vibe/core/process.d

```diff
@@ -0,0 +1,758 @@
+/**
+    Functions and structures for dealing with subprocesses and pipes.
+
+    This module is modeled after std.process, but provides a fiber-aware
+    alternative to it. All blocking operations will yield the calling fiber
+    instead of blocking it.
+*/
+module vibe.core.process;
+
+public import std.process : Pid, Redirect;
+static import std.process;
+
+import core.time;
+import std.array;
+import std.typecons;
+import std.exception : enforce;
+import std.algorithm;
+import eventcore.core;
+import vibe.core.path;
+import vibe.core.log;
+import vibe.core.stream;
+import vibe.internal.async;
+import vibe.internal.array : BatchBuffer;
+import vibe.core.internal.release;
+
+@safe:
+
+/**
+    Register a process with vibe for fibre-aware handling. This process can be
+    started from anywhere including external libraries or std.process.
+
+    Params:
+        pid = A Pid or OS process handle
+*/
+Process registerProcess(Pid pid)
```

Comment:
My initial thought for the name was `openProcess(by(P)ID)`, probably because that is what the Windows API uses. Register process may sound a bit like it would statically register the process somewhere instead of just returning a handle. What do you think?

Reply:
Maybe `adoptProcess` would be better and more in-line with eventcore?

Reply:
Thought about that, too, but dismissed it at the time, because it also sounded a bit like it would do something with the process as opposed to just creating a "handle" for it. But apart from eventcore using it, another argument for "adopt" would be consistency with a possible analoguous socket API - `openSocket` would be rather misleading there.

I think the mentioned issue can be fixed quite easily, though, by including "ID" in the name (i.e. `adoptProcessID` or `adoptPID`), so "adopt" sounds fine after all.

Reply:
Good idea, I'll make the change